src/aircloak/JsonApi/JsonApiClient.cs

```diff
@@ -135,15 +135,23 @@ public async Task<DataSourceCollection> GetDataSources(CancellationToken cancell
                 Converters = { new JsonArrayConverter<IQuerySpec<TRow>, TRow>(querySpec) },
             };
 
+            var queryCompleted = false;
+            var speed = 16;
+
             try
             {
                 while (true)
                 {
+                    await Task.Delay(pollFrequency / speed, cancellationToken);
+
+                    cancellationToken.ThrowIfCancellationRequested();
+
```

Comment:
Why did you move the delay to the beginning of the loop? Surely this means there's a (short) delay before even sending the request for the first time?

Reply:
fixed with 6f0f7afeb172bae2cdbc80959addbd0b118eb40b